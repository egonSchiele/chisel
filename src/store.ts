import * as toolkitRaw from "@reduxjs/toolkit";
import { librarySlice } from "./reducers/librarySlice";

// @ts-ignore
const { configureStore } = toolkitRaw.default ?? toolkitRaw;

export const store = configureStore({
  reducer: {
    library: librarySlice.reducer,
  },
  middleware: (getDefaultMiddleware) =>
  getDefaultMiddleware({
    serializableCheck: false,
  }),
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
