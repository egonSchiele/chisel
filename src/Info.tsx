import React from "react";
import { useSelector } from "react-redux";
import { syllable } from "syllable";
import { RootState } from "./store";
import { getCharacters } from "./reducers/librarySlice";

const countSyllables = (text: string) => {
  try {
    return syllable(text);
  } catch (error) {
    console.error("Error counting syllables:", error);
    return 0;
  }
};

function CharacterInfo() {
  const editor = useSelector((state: RootState) => state.library.editor);
  const characters = useSelector(getCharacters);
  if (!characters) return null;
  if (editor && editor.selectedText && editor.selectedText.contents) {
    const character = characters.find(
      (character) =>
        character.name.toLowerCase() ===
        editor.selectedText.contents.toLowerCase()
    );
    if (character) {
      return (
        <div className="mt-sm bg-gray-200 dark:bg-gray-700 rounded-md p-sm">
          {character.imageUrl && character.imageUrl.trim() !== "" && (
            <img
              src={character.imageUrl}
              alt={character.name}
              className="rounded-full mx-auto mb-sm"
            />
          )}
          <p className="text-lg font-semibold">{character.name}</p>
          <p className="text-sm">{character.description}</p>
        </div>
      );
    }
  }
  return null;
}
export default function Info({ text }) {
  const word_count = text.trim().split(/\s+/).length;
  const syllable_count = countSyllables(text.trim());
  return (
    <div className="text-sm xl:text-md">
      <p>
        {word_count} <span className="text-gray-400">words</span>
      </p>
      <p>
        {Math.floor(word_count * (4 / 3))}{" "}
        <span className="text-gray-400">tokens (estimate)</span>
      </p>
      {/*  <p>
        {syllable_count} <span className="text-gray-400">syllables</span>
      </p> */}
      <CharacterInfo />
    </div>
  );
}
