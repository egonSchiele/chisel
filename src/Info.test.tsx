// @ts-nocheck
import renderer from "react-test-renderer";
import React from "react";
import Info from "./Info";
import { Provider } from "react-redux";

import { store } from "./store";

it("shows the info panel", () => {
  const component = renderer.create(
    <Provider store={store}>
      <Info text="hi there syllables" />
    </Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
