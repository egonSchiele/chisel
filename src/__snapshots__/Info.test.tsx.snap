// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows the info panel 1`] = `
<div
  className="text-sm xl:text-md"
>
  <p>
    3
     
    <span
      className="text-gray-400"
    >
      words
    </span>
  </p>
  <p>
    5
     
    <span
      className="text-gray-400"
    >
      syllables
    </span>
  </p>
  <p>
    4
     
    <span
      className="text-gray-400"
    >
      tokens (estimate)
    </span>
  </p>
  <p>
    1 min read
     
    <span
      className="text-gray-400"
    />
  </p>
</div>
`;
